src/main/java/ch/wisv/areafiftylan/products/model/Order.java

```diff
@@ -34,89 +36,72 @@
     @Id
     @GeneratedValue
     @JsonView(View.OrderOverview.class)
+    @Getter
     private Long id;
 
     @OneToMany(cascade = CascadeType.MERGE, targetEntity = Ticket.class, fetch = FetchType.EAGER)
     @JsonView(View.OrderOverview.class)
+    @Getter
     private Set<Ticket> tickets;
 
     @JsonView(View.OrderOverview.class)
+    @Getter
+    @Setter
     private OrderStatus status;
 
     @JsonView(View.OrderOverview.class)
     @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
+    @Getter
     private LocalDateTime creationDateTime;
 
-    @JsonView(View.OrderOverview.class)
-    private float amount;
-
     /**
      * This String can be used to store an external reference. Payment providers often have their own id.
      */
     @JsonView(View.OrderOverview.class)
+    @Getter
+    @Setter
     private String reference;
 
     @ManyToOne(cascade = CascadeType.MERGE)
     @JsonView(View.OrderOverview.class)
-    private User user;
+    @Getter
+    private User user = null;
 
-    public Order(User user) {
-        this.user = user;
-        status = OrderStatus.CREATING;
+    public Order() {
+        status = OrderStatus.ANONYMOUS;
         creationDateTime = LocalDateTime.now();
         this.tickets = new HashSet<>();
     }
 
-    public Order() {
-        //JPA only
-    }
-
-    public Long getId() {
-        return id;
-    }
-
-    public Set<Ticket> getTickets() {
-        return tickets;
+    public Order(User user) {
+        this();
+        this.user = user;
+        this.status = OrderStatus.ASSIGNED;
     }
 
     public boolean addTicket(Ticket ticket) {
-        amount += ticket.getPrice();
         return tickets.add(ticket);
     }
 
     public void clearTickets() {
         tickets.clear();
     }
 
-    public OrderStatus getStatus() {
-        return status;
-    }
-
-    public LocalDateTime getCreationDateTime() {
-        return creationDateTime;
-    }
-
-    public String getReference() {
-        return reference;
-    }
-
-    public User getUser() {
-        return user;
-    }
-
-    public void setStatus(OrderStatus status) {
-        this.status = status;
-    }
-
-    public void setReference(String reference) {
-        this.reference = reference;
-    }
-
+    @JsonView(View.OrderOverview.class)
     public float getAmount() {
         float price = 0F;
         for (Ticket ticket : this.tickets) {
             price += ticket.getPrice();
         }
         return price;
     }
+
+    public void setUser(User user) {
+        this.user = user;
+        if (user != null) {
```

Comment:
We could also have check in here that throws an exception? An `assert(...)` is also possible. This looks quite weird to me, checking null-ness on a setter method.